lib/ansible/modules/rpm_key.py

```diff
@@ -105,35 +106,37 @@ def __init__(self, module):
         state = module.params['state']
         key = module.params['key']
         fingerprint = module.params['fingerprint']
+
         if fingerprint:
-            fingerprint = fingerprint.replace(' ', '').upper()
+            fingerprint = [fprnt.replace(' ', '').upper() for fprnt in fingerprint]
 
         self.gpg = self.module.get_bin_path('gpg')
         if not self.gpg:
             self.gpg = self.module.get_bin_path('gpg2', required=True)
 
+        # Download the keyfile if a URL
         if '://' in key:
             keyfile = self.fetch_key(key)
-            keyid = self.getkeyid(keyfile)
+            keyids = self.getkeyids(keyfile)
             should_cleanup_keyfile = True
         elif self.is_keyid(key):
-            keyid = key
+            keyids = [key]
         elif os.path.isfile(key):
             keyfile = key
-            keyid = self.getkeyid(keyfile)
+            keyids = self.getkeyids(keyfile)
         else:
             self.module.fail_json(msg="Not a valid key %s" % key)
-        keyid = self.normalize_keyid(keyid)
+        keyids = self.normalize_keyids(keyids)
 
         if state == 'present':
-            if self.is_key_imported(keyid):
+            if self.is_key_imported(keyids):
                 module.exit_json(changed=False)
             else:
                 if not keyfile:
                     self.module.fail_json(msg="When importing a key, a valid file must be given")
                 if fingerprint:
                     has_fingerprint = self.getfingerprint(keyfile)
-                    if fingerprint != has_fingerprint:
+                    if set(fingerprint) != set(has_fingerprint):
```

Comment:
Should this be something along the lines of
```suggestion
                    if any(fgrprnt not in computed_fingerprint for fgrprnt in fingerprints):
```
?

Perhaps, this entire expression could be assigned to a variable, so its intended purpose could be labeled via a variable name?